docs/learn/documentation/versioned/jobs/configuration-table.html

```diff
@@ -822,6 +828,40 @@
                 </tr>
 
                 <tr>
+                    <th colspan="3" class="section" id="streams"><a href="../container/streams.html">Streams</a></th>
+                </tr>
+
+                <tr>
+                    <td class="property" id="streams-system">streams.<span class="stream">stream-id</span>.samza.system</td>
+                    <td class="default"></td>
+                    <td class="description">
+                        The <span class="system">system-name</span> of the system on which this stream will be accessed.
+                        This property binds the stream to one of the systems defined with the property
+                        systems.<span class="system">system-name</span>.samza.factory. <br>
+                        If this property isn't specified, it is inherited from job.default.system.
+                    </td>
+                </tr>
+
+                <tr>
+                    <td class="property" id="streams-physical-name">streams.<span class="stream">stream-id</span>.samza.physical.name</td>
```

Comment:
Yes it's intentional and in a separate task we'll go deprecate the SystemStream approach to configurations. That will be a large effort, so I didn't pursue it here. For now, the new streams.streamName.* approach takes precedence. When we remove the legacy stream configs, the documentation for those properties will be moved here with the other streams properties.

We're doing this because the "stream" in SystemStream could be any physical name like an HDFS file URN. Obviously that will be hard to parse in a config file that has systems.sysName.streams.streamName.* 
So that path is already broken for streams that have special characters. StreamId fixes that by decoupling the logical identifier used by samza from the physical/remote identifier used by the system. That way we can enforce no special characters for the streamId so we can use it as a key in the configs.

System level properties will still be set at the system level for now. i.e. systems.systemName.*. In the future, I'd like to define template streams which better serve this purpose. Something like 
templates.streams.myAvroTemplate.samza.msg.serde=avro
streams.myStream.samza.template=myAvroTemplate

Then you could define a bunch of properties on the template and apply the template as a set of defaults to each stream. Just a rough idea for now.